packages/dds/matrix/src/serialization.ts

```diff
@@ -3,45 +3,29 @@
  * Licensed under the MIT License.
  */
 
-import {
-    Serializable,
-    IFluidDataStoreRuntime,
-    IChannelStorageService,
-} from "@fluidframework/datastore-definitions";
+import { Serializable, IChannelStorageService } from "@fluidframework/datastore-definitions";
 import { FileMode, TreeEntry } from "@fluidframework/protocol-definitions";
-import { IFluidHandle } from "@fluidframework/core-interfaces";
+import { IFluidHandle, IFluidSerializer } from "@fluidframework/core-interfaces";
 import { fromBase64ToUtf8 } from "@fluidframework/common-utils";
 
-export function serializeBlob(
-    runtime: IFluidDataStoreRuntime,
+export const serializeBlob = (
     handle: IFluidHandle,
     path: string,
     snapshot: Serializable,
-) {
-    const serializer = runtime.IFluidSerializer;
-
-    return {
+    serializer: IFluidSerializer,
+) => ({
         mode: FileMode.File,
         path,
         type: TreeEntry.Blob,
         value: {
-            contents: serializer !== undefined
-                ? serializer.stringify(snapshot, handle)
-                : JSON.stringify(snapshot),
+            contents: serializer.stringify(snapshot, handle),
             encoding: "utf-8",
         },
-    };
-}
+    });
 
-export async function deserializeBlob(runtime: IFluidDataStoreRuntime, storage: IChannelStorageService, path: string) {
+export async function deserializeBlob(storage: IChannelStorageService, path: string, serializer: IFluidSerializer) {
     const handleTableChunk = await storage.read(path);
     const utf8 = fromBase64ToUtf8(handleTableChunk);
```

Comment:
This is general comment, not directly related to your change: We should stop using "utf8" as much as possible, and use "string" or "content" nomenclature. Because we have no clue how stings are represented in memory of JS engine.

Reply:
Good to know. I will keep it in mind.